web-services-app/src/main/java/edu/unc/lib/boxc/web/services/rest/modify/AccessSurrogateController.java

```diff
@@ -0,0 +1,166 @@
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+
+import static edu.unc.lib.boxc.auth.fcrepo.services.GroupsThreadStore.getAgentPrincipals;
+import static edu.unc.lib.boxc.model.api.ids.RepositoryPathConstants.HASHED_PATH_DEPTH;
+import static edu.unc.lib.boxc.model.api.ids.RepositoryPathConstants.HASHED_PATH_SIZE;
+import static edu.unc.lib.boxc.model.fcrepo.ids.RepositoryPaths.idToPath;
+import static edu.unc.lib.boxc.operations.jms.accessSurrogates.AccessSurrogateRequest.DELETE;
+import static edu.unc.lib.boxc.operations.jms.accessSurrogates.AccessSurrogateRequest.SET;
+import static org.apache.commons.io.FileUtils.copyInputStreamToFile;
+import static org.apache.commons.lang3.StringUtils.containsIgnoreCase;
+
+/**
+ * Controller for handling access surrogate requests. This may include things like
+ * setting, replacing, or deleting an access surrogate
+ */
+@Controller
+public class AccessSurrogateController {
+    private static final Logger log = LoggerFactory.getLogger(AccessSurrogateController.class);
+    public static final String ACTION = "action";
+    @Autowired
+    private Path accessSurrogateTempPath;
+    @Autowired
+    private AccessControlService aclService;
+    @Autowired
+    private RepositoryObjectLoader repositoryObjectLoader;
+    @Autowired
+    private AccessSurrogateRequestSender accessSurrogateRequestSender;
+
+    @PostMapping(value = "/edit/accessSurrogate/{id}")
+    @ResponseBody
+    public ResponseEntity<Object> setAccessSurrogate(@PathVariable("id") String pidString,
+                                                        @RequestParam("file") MultipartFile surrogateFile) {
+        Map<String, Object> result = new HashMap<>();
+        PID pid = PIDs.get(pidString);
+        AccessGroupSet principals = getAgentPrincipals().getPrincipals();
+        aclService.assertHasAccess("Insufficient permissions to set access surrogates for " + pidString,
+                pid, principals, Permission.editDescription);
+
+        // check if object is a file
+        var repositoryObject = repositoryObjectLoader.getRepositoryObject(pid);
+        if (!(repositoryObject instanceof FileObject)) {
+            throw new InvalidOperationForObjectType("Cannot set access surrogate of object " +
+                    pidString + ", as only FileObjects have access surrogates");
+        }
+
+        var request = buildRequest(SET, pidString);
+
+        // uploaded file must be an image
+        var mimeType = surrogateFile.getContentType();
+        if (!containsIgnoreCase(mimeType, "image")) {
+            log.error("Uploaded file for collection {} is not an image file", pidString);
```

Comment:
Might make sense for this to be a `warn` since it is user input error and doesn't generally require developer followup